presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -97,8 +105,13 @@ public void process(Iterable<QueryExecution> queries)
         }
         clusterMemoryUsageBytes.set(totalBytes);
 
-        updateNodes();
-        updatePools();
+        ImmutableList<QueryInfo> queryInfos = ImmutableList.copyOf(queries).stream()
+                .map(QueryExecution::getQueryInfo)
+                .collect(toImmutableList());
+
+        updatePools(queryInfos);
```

Comment:
IIRC, building the query info can be quite expensive.  You may want to just get the memory pool id directly from the query execution.  Also I'm not sure I like the idea of passing the query info all the way to the ClusterMemoryPool, just so it can count the queries... maybe count the queries for them and pass in the count?